### PR TITLE
sql/pgwire: remove `pgType.preferredFormat`

### PR DESCRIPTION
**Pull request by tamird**
Save some allocations while we're at it.

cc @mjibson @bdarnell @petermattis

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3484)

### CODE REVIEW

**Comment by petermattis** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/pgwire/v3.go, line 321 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3484#-K5lWS7AfOD7hkDX06b-:-K5lWS7AfOD7hkDX06b0:134117032) ([raw file](https://github.com/cockroachdb/cockroach/blob/72d3ec15a0340f67cd33dbace1e5571ceef0b954/sql/pgwire/v3.go#L321)):*
Looks like `sendResponse()` is only ever called with `formatCodes == nil`. Is this now detritus?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3484)*

**Comment by tamird** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/pgwire/v3.go, line 321 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3484#-K5lWS7AfOD7hkDX06b-:-K5lZ9O3Nxris_0zJVTa:609061954) ([raw file](https://github.com/cockroachdb/cockroach/blob/72d3ec15a0340f67cd33dbace1e5571ceef0b954/sql/pgwire/v3.go#L321)):*
That's correct, but this function will be used in the prepared statements branch.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3484)*

**Comment by mjibson** [comment]
LGTM

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3484#-:-K5lb31cpAOFLwIjRuwD:676834233)*

File: sql/pgwire/v3.go [hunk 1]
```diff
@@ -324,7 +317,11 @@ func (c *v3Conn) sendResponse(resp driver.Response, formatCodes []formatCode) er
 				c.writeBuf.initMsg(serverMsgDataRow)
 				c.writeBuf.putInt16(int16(len(row.Values)))
 				for i, col := range row.Values {
-					switch formatCode := formatCodes[i]; formatCode {
+					fmtCode := formatText
+					if formatCodes != nil {
+						fmtCode = formatCodes[i]
```
Comment (bdarnell) [inline sql/pgwire/v3.go]: The `formatCodes` list may contain a single value, in which case that value is applied to all parameters (not sure when anyone would do that, but it's in the protocol spec).

Comment (mjibson) [inline sql/pgwire/v3.go]: We have code that handles that in the prepared statement branch.

**Pull request merged**